crates/ruff_python_formatter/src/statement/stmt_class_def.rs

```diff
@@ -36,71 +37,76 @@ impl FormatNodeRule<StmtClassDef> for FormatStmtClassDef {
         }
         .fmt(f)?;
 
-        write!(f, [text("class"), space(), name.format()])?;
+        if SuppressionKind::has_skip_comment(trailing_definition_comments, f.context().source()) {
+            SuppressedClauseHeader::Class(item).fmt(f)?;
+        } else {
+            write!(f, [text("class"), space(), name.format()])?;
```

Comment:
The pattern is always the same. Test if there's a trailing skip comment and, if so, call `SuppressedClauseHeader`. The enclosing node remains responsible for formatting the body and the trailing comments.